x-pack/filebeat/input/internal/private/private_test.go

```diff
@@ -36,6 +39,34 @@ var redactTests = []redactTest{
 			"not_secret": "2",
 		},
 	},
+	{
+		name: "map_string_replacer",
+		in: map[string]any{
+			"private":    "secret",
+			"secret":     "this is a secret",
+			"not_secret": "this is not",
+		},
+		replacers: []Replacer{NewStringReplacer("REDACTED")},
+		want: map[string]any{
+			"private":    "secret",
+			"secret":     "REDACTED",
+			"not_secret": "this is not",
+		},
+	},
+	{
+		name: "map_string_custom_replacer",
+		in: map[string]any{
+			"private":    "secret",
+			"secret":     "this is a secret",
+			"not_secret": "this is not",
+		},
+		replacers: []Replacer{func(s string) string { return strings.Repeat("*", len(s)) }},
+		want: map[string]any{
+			"private":    "secret",
+			"secret":     "****************", // Same length as original.
+			"not_secret": "this is not",
+		},
```

Comment:
I know this is just a test and it's up to the caller, but I think a fixed length would be better, in order to not leak the secret length.

In a UI, showing a masked secret's length helps with data entry. Here it may just reduce the search space considerably.

Reply:
That's why this is not an exported function. It is merely here to show that the output depends on the input. The intended use case is to allow cryptographically traceable secret leakage.